src/Symfony/Component/HttpKernel/Profiler/ProfilerStorageInterface.php

```diff
@@ -29,11 +29,12 @@ interface ProfilerStorageInterface
     /**
      * Finds profiler tokens for the given criteria.
      *
-     * @param int|null $limit The maximum number of tokens to return
-     * @param int|null $start The start date to search from
-     * @param int|null $end   The end date to search to
+     * @param int|null      $limit  The maximum number of tokens to return
+     * @param int|null      $start  The start date to search from
+     * @param int|null      $end    The end date to search to
+     * @param \Closure|null $filter A filter to apply on the list of tokens
      */
-    public function find(?string $ip, ?string $url, ?int $limit, ?string $method, int $start = null, int $end = null): array;
+    public function find(?string $ip, ?string $url, ?int $limit, ?string $method, int $start = null, int $end = null /* , \Closure $filter = null*/): array;
```

Comment:
I figured out why we need to change this method: to filter by request type so that the limit argument makes sense.
Instead of adding a new `$virtualType` argument, I pushed a more generic `$filter` one.